internal/api/ui/login/external_provider_handler.go

```diff
@@ -501,10 +501,22 @@ func (l *Login) checkAutoLinking(w http.ResponseWriter, r *http.Request, authReq
 	if err != nil {
 		return false
 	}
-	l.renderLinkingUserPrompt(w, r, authReq, user, nil)
+	l.autoLinkUser(w, r, authReq, user)
 	return true
 }
 
+func (l *Login) autoLinkUser(w http.ResponseWriter, r *http.Request, authReq *domain.AuthRequest, user *query.NotifyUser) {
+	err := l.authRepo.SelectUser(r.Context(), authReq.ID, user.ID, authReq.AgentID)
+	if err != nil {
```

Comment:
Should we make a one-liner out of the error condition?

Reply:
more importantly i added a `return` on error

Reply:
Good point, missed that as the renderError function was there.